services/core/MasterDriverAgent/master_driver/agent.py

```diff
@@ -467,66 +490,104 @@ def set_override_off(self, pattern):
     # Get a list of all the devices with override condition.
     @RPC.export
     def get_override_devices(self):
-        devices = []
-        devices.extend(self._override_devices)
-        return devices
+        return list(self._override_devices)
 
     # Clear all overrides
     @RPC.export
     def clear_overrides(self):
         # Cancel all pending override timer events
-        for evt in self._override_interval_events.items():
+        for pattern, evt in self._override_interval_events.items():
             if evt is not None:
-                evt.cancel()
+                evt[0].cancel()
         self._override_interval_events.clear()
         self._override_devices.clear()
         self._override_patterns.clear()
+        self.vip.config.set("override_patterns", {})
 
     # Get a list of all the devices with override condition.
     @RPC.export
     def get_override_patterns(self):
-        patterns = []
-        patterns.extend(self._override_patterns)
-        return patterns
+        return list(self._override_patterns)
 
     def _set_override_off(self, pattern):
         pattern = pattern.lower()
+        config_update = False
+
         # If pattern exactly matches
         if pattern in self._override_patterns:
             self._override_patterns.discard(pattern)
+            # Cancel any pending override events
+            self._cancel_override_events(pattern)
+            config_update = True
         else:
             _log.debug("Pattern did not match!")
 
         self._override_devices.clear()
-        #Cancel any pending override events
-        self._cancel_override_events(pattern)
+        patterns = dict()
         #Build override devices list again
         for pat in self._override_patterns:
             for device in self.instances.keys():
                 device = device.lower()
                 if fnmatch.fnmatch(device, pat):
                     self._override_devices.add(device)
-
-    def _update_override_interval(self, interval, topic):
-        override_start = utils.get_aware_utc_now()
-        override_end = override_start + timedelta(seconds=interval)
-        try:
-            self._override_interval_events[topic].cancel()
-        except KeyError:
-            _log.debug("MASTER DRIVER override interval event key error {}".format(topic))
-            pass
-        self._override_interval_events[topic] = self.core.schedule(override_end, self._cancel_override, topic)
+            if config_update:
+                if self._override_interval_events[pat] is None:
+                    patterns[pat] = None
+                else:
+                    evt, end_time = self._override_interval_events[pat]
+                    patterns[pat] = end_time.isoformat()
```

Comment:
Everywhere a datetime object is converted into a string please use volttron.platform.agent.utils.format_timestamp.